GliaWidgets/Sources/Coordinators/Chat/ChatCoordinator.swift

```diff
@@ -51,15 +53,84 @@ class ChatCoordinator: SubFlowCoordinator, FlowCoordinator {
         self.isWindowVisible = isWindowVisible
         self.startAction = startAction
         self.environment = environment
+        self.shouldStartWithTranscript = shouldStartWithTranscript
     }
 
     func start() -> ChatViewController {
         let viewController = makeChatViewController()
         return viewController
     }
 
-    // swiftlint:disable function_body_length
     private func makeChatViewController() -> ChatViewController {
+        let model: SecureConversations.ChatWithTranscriptModel =
+            shouldStartWithTranscript ? .transcript(transcriptModel()) : .chat(chatModel())
+        return ChatViewController(viewModel: model, viewFactory: viewFactory)
+    }
+
+    private func presentMediaPickerController(
+        with pickerEvent: ObservableValue<MediaPickerEvent>,
+        mediaSource: MediaPickerViewModel.MediaSource,
+        mediaTypes: [MediaPickerViewModel.MediaType]
+    ) {
+        let viewModel = MediaPickerViewModel(
+            pickerEvent: pickerEvent,
+            mediaSource: mediaSource,
+            mediaTypes: mediaTypes
+        )
+        viewModel.delegate = { [weak self] event in
+            switch event {
+            case .finished:
+                self?.mediaPickerController = nil
+            }
+        }
+
+        let controller = MediaPickerController(viewModel: viewModel)
+        mediaPickerController = controller
+        controller.viewController { [weak self] viewController in
+            self?.navigationPresenter.present(viewController)
+        }
+    }
+
+    private func presentFilePickerController(with pickerEvent: ObservableValue<FilePickerEvent>) {
+        let viewModel = FilePickerViewModel(pickerEvent: pickerEvent)
+        viewModel.delegate = { [weak self] event in
+            switch event {
+            case .finished:
+                self?.filePickerController = nil
+            }
+        }
+
+        let controller = FilePickerController(viewModel: viewModel)
+        filePickerController = controller
+        navigationPresenter.present(controller.viewController)
+    }
+
+    private func presentQuickLookController(with file: LocalFile) {
```

Comment:
what is QuickLookController?

Reply:
A wrapper class around QLPreviewController. Used for file previews.